src/apprt/gtk/menu.zig

```diff
@@ -0,0 +1,125 @@
+const std = @import("std");
+
+const apprt = @import("../../apprt.zig");
+const App = @import("App.zig");
+const Window = @import("Window.zig");
+const Surface = @import("Surface.zig");
+const Builder = @import("Builder.zig");
+
+const gtk = @import("gtk");
+const gdk = @import("gdk");
+const gio = @import("gio");
+const gobject = @import("gobject");
+
+const log = std.log.scoped(.gtk_menu);
+
+pub fn Menu(
+    comptime T: type,
+    comptime variant: []const u8,
+    comptime style: enum { popover_menu, popover_menu_no_arrow },
+) type {
+    return struct {
+        const Self = @This();
+        const MenuWidget = switch (style) {
+            .popover_menu => gtk.PopoverMenu,
+            .popover_menu_no_arrow => gtk.PopoverMenu,
+        };
+
+        parent: *T,
+        menu_widget: *MenuWidget,
+
+        pub fn init(self: *Self) void {
+            const name = switch (T) {
+                Window => "window",
+                Surface => "surface",
+                else => unreachable,
+            };
+            const parent: *T = @alignCast(@fieldParentPtr(variant, self));
```

Comment:
I implore us to reconsider `@fieldParentPtr` if we can. The use of it is a huge smell that something is wrong. I noted this in another PR. I don't have enough context on the gobject work to know if we must do this, but let's think about it. I'll take a closer look another time (feel free to not block this merge on it).

Reply:
Removed.